project1/src/main/java/uk/co/cpsd/javaproject1/World.java

```diff
@@ -74,6 +116,29 @@ public void tick() {
 
     public void moveAnimals(){
         animals.forEach(animal-> animal.act(this));
+    }
+
+    public void handleReproduction(){
 
+        Set<Integer> alreadyUsed=new HashSet<>();
+        List<Animal> newBabies=new ArrayList<>();
+        for(Animal animal1:animals){
+            if (alreadyUsed.contains(animal1.getId())) continue;
+            for(Animal animal2:animals){
+                if(animal1==animal2 || alreadyUsed.contains(animal2.getId())) continue;
+
+                Animal baby= animal1.tryReproduceWith(animal2);
+                if(baby!=null){
+                    newBabies.add(baby);
+                    System.out.println("=================Bbay was born==================");
```

Comment:
It looks like this algorithm works.  It also feels very inefficient.  Go with this for this PR merge, but I think you need to think about how to do this more efficiently going forward.  It will probably help to think about a more complex animal.  So an animal that looks at its surroundings and chooses and action.  Move/Eat/Reproduce/Flee.  If you have a concept of nearness to an animal and how it might react, we suddenly get a whole new level of simulation.  It's complicated, it is going to test you, but it takes the software to a new level.